### DISCUSSION TITLE
BufferedImage option for WrappedServerPing.CompressedImage

### DISCUSSION THREAD

**Discussion by Andre601** [dmulloy2/ProtocolLib#1862] [Ideas]
From what I can tell is there no option to simply provide a `BufferedImage` for a `WrappedServerPing.CompressedImage` instance to be created.

This should be considered to be implemented, as every server and proxy platform seems to provide an option to create a sort of Favicon instance from a BufferedImage itself.

The fact that ProtocolLib seems to be the only case to not offer this makes it a lot more difficult to write a central system to allow stuff such as creation of favicons for multiple platforms.
I myself encountered this exact issue. While Paper, BungeeCord and Velocity have an option to create a Favicon, do I have to use a byte array for ProtocolLib, to achieve the same goal for Spigot.

So, if doable should it be considered to implement a method like `fromPng(BufferedImage)` for `WrappedServerPing.CompressedImage` creation.

Maybe also consider removing a few other methods? Like what situation justifies the usage of a(n encoded) String?

**Comment by derklaro** [comment]
Isn't `CompressedImage#fromPng(RenderedImage image)` doing the trick?

**Reply by Andre601** [reply]
Yeah, that does the trick.
